apps/src/storage/dataBrowser/DataTable.jsx

```diff
@@ -258,35 +259,43 @@ class DataTable extends React.Component {
                   }
                   isEditing={editingColumn === columnName}
                   isPending={this.state.pendingColumn === columnName}
+                  readOnly={this.props.readOnly}
                   renameColumn={this.renameColumn}
                 />
               ))}
-              <th style={styles.addColumnHeader}>
-                {this.state.pendingAdd ? (
-                  <FontAwesome icon="spinner" className="fa-spin" />
-                ) : (
-                  <FontAwesome
-                    id="addColumnButton"
-                    icon="plus"
-                    style={styles.plusIcon}
-                    onClick={this.addColumn}
-                  />
-                )}
-              </th>
-              <th style={dataStyles.headerCell}>Actions</th>
+              {!this.props.readOnly && (
+                <th style={styles.addColumnHeader}>
+                  {this.state.pendingAdd ? (
+                    <FontAwesome icon="spinner" className="fa-spin" />
+                  ) : (
+                    <FontAwesome
+                      id="addColumnButton"
+                      icon="plus"
+                      style={styles.plusIcon}
+                      onClick={this.addColumn}
+                    />
+                  )}
+                </th>
+              )}
+              {!this.props.readOnly && (
```

Comment:
Can both of these `{!this.props.readOnly &&` be combined into one?

Reply:
Combining here and in EditTableRow requires using React.Fragment to group the elements. No strong opinion on whether or not this is cleaner than doing two `!this.props.readOnly &&` checks. Which do you prefer?

Reply:
Edit: We can't actually use React.Fragment 🤦‍♀️ (requires React 16.2) So I think this is not possible to combine (would require wrapping in a div in order to meet the one element requirement, but then would be invalid HTML because it's inside a table)

Reply:
Oh I see, I forgot that we can't just wrap them in a div because it's a table! Yup, nvm.